static/src/javascripts/projects/commercial/modules/comment-adverts.spec.ts

```diff
@@ -133,10 +136,10 @@ describe('maybeUpgradeSlot', () => {
         const advert = {
```

Comment:
Could you not declare `advert: Advert` here?

Reply:
The problem here is we are creating a type which contains a subset of the `Advert` props for testing convenience.

But then passing it to the implementation which is expecting `Advert` even though it has missing properties.

I tried `Advert`, `Partial<Advert>` and extending `Advert` but ran into the same/further problems when using the object with the real implementation which is expecting `Advert`.

`Partial` does type checking on the properties but you still need to `(advert as Advert)` whenever using the object.

Given its a test convenience and it's not actually a real `Advert` this seemed like the easiest workaround.

I have however updated the test to use a function `createTestAdvert` which creates a `Partial` and then returns `as Advert` which shifts the problem to one function.

```js
const createTestAdvert = (testAdvert: Partial<Advert>): Advert =>
    ({ ... testAdvert } as Advert);
```

Reply:
Yeah I had the same issues with Advert.test.ts - this kind of issue must come up a lot, maybe someone has found a better way to deal with creating partial objects as mocks when testing but keeping TS happy